infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -119,6 +120,22 @@ public StreamLogFiles(ServerContext serverContext, boolean noVerify) {
         }
     }
 
+    /**
+     * Create stream log directory if not exists
+     */
+    private void initStreamLogDirectory() {
+        if (logDir.toFile().exists()) {
+            log.info("Log directory already exists: {}", logDir);
+            return;
+        }
+
+        try {
+            Files.createDirectories(logDir);
```

Comment:
I think you need to fsync the parent directory after you create the log directory (i.e use Persistence.syncDirectory) , or maybe just the directory itself since we "only" support ext4: 

https://github.com/torvalds/linux/blob/master/fs/ext4/fsync.c#L122

Reply:
Thanks, will apply it in the next PR